File: cfgs/pythoncfgs/STOOGE_SORT.py
Nodes
1;conditional;foo_1 >= foo_2;8;
2;conditional;foo_1[foo_2] > foo_3[foo_4];10;
3;statement;foo_1 - foo_2 + 1 > 2;14;
4;statement;foo_1 = fun_1((foo_2 - foo_3 + 1) / 3);15;
5;statement;fun_1(foo_1, foo_2, foo_3 - foo_4);16;
6;statement;fun_1(foo_1, foo_2 + foo_3, foo_4);17;
7;statement;fun_1(foo_1, foo_2, foo_3 - foo_4);18;
8;statement;foo_1 = foo_2[foo_3];11;
9;statement;foo_1[foo_2] = foo_3[foo_4];12;
10;statement;foo_1[foo_2] = foo_3;13;
11;statement;return;9;
12;exit;;
Edges
0;2;11;
1;3;8;
2;4;
3;5;
4;6;
5;7;
6;12;
7;9;
8;10;
9;3;
10;12;
11;

File: cfgs/java_cfgs_foo/STOOGE_SORT.java
Nodes
1;special;;;
2;conditional;foo_1 >= foo_2;14;
3;statement;return;14;
4;exit;;;
5;conditional;foo_1[foo_2] > foo_3[foo_4];15;
6;statement;int t = foo_1[foo_2];16;
7;statement;foo_1[foo_2] = foo_3[foo_4];17;
8;statement;foo_1[foo_2] = foo_3;18;
9;conditional;((foo_1 - foo_2) + 1) > 2;20;
10;statement;int t = ((foo_1 - foo_2) + 1) / 3;21;
11;statement;fun_1(foo_1, foo_2, foo_3 - foo_4);22;
12;statement;fun_1(foo_1, foo_2 + foo_3, foo_4);23;
13;statement;fun_1(foo_1, foo_2, foo_3 - foo_4);24;
Edges
0;2;
1;3;5;
2;4;
3;
4;6;9;
5;7;
6;8;
7;9;
8;10;4;
9;11;
10;12;
11;13;
12;4;
